website/content/docs/connect/observability/grafanadashboards/servicedashboard.mdx

```diff
@@ -0,0 +1,153 @@
+```
+
+### Total failed request rate
+
+**Description:** This stat tracks the rate of failed requests (4xx and 5xx errors) for the selected service. It helps operators quickly identify if there are issues with client requests or server errors for a specific service.
+
+```promql
+sum(irate(envoy_cluster_upstream_rq_xx{envoy_response_code_class=~"4|5", consul_destination_service=~"$service"}[10m])) / sum(irate(envoy_cluster_upstream_rq_xx{consul_destination_service=~"$service"}[10m]))
+```
+
+### Average request response time in milliseconds
+
+**Description:** This gauge displays the average response time for requests to the selected service, providing an overview of the service's performance and responsiveness.
+
+```promql
+sum(rate(envoy_cluster_upstream_rq_time_sum{consul_destination_service=~"$service"}[10m])) / sum(rate(envoy_cluster_upstream_rq_total{consul_destination_service=~"$service"}[10m]))
+```
+
+### Total failed requests
+
+**Description:** This gauge tracks the total number of failed requests over a 10-minute window, categorized by service. It allows for easy identification of services that are experiencing high failure rates.
+
+```promql
+sum(increase(envoy_cluster_upstream_rq_xx{envoy_response_code_class=~"4|5", consul_destination_service=~"$service"}[10m])) by(local_cluster)
+```
+
+### Dataplane latency
+
+**Description:** This stat tracks the dataplane latency percentiles (p50, p75, p90, p99.9) for the selected service. It gives detailed insights into the distribution of latency within the service's request handling, helping identify performance bottlenecks.
+
+```promql
+histogram_quantile(0.50, sum by(le) (rate(envoy_cluster_upstream_rq_time_bucket{kubernetes_namespace=~"$namespace", local_cluster=~"$service"}[5m])))
+```
+
+```promql
+histogram_quantile(0.75, sum by(le) (rate(envoy_cluster_upstream_rq_time_bucket{kubernetes_namespace=~"$namespace", local_cluster=~"$service"}[5m])))
+```
+
+```promql
+histogram_quantile(0.90, sum by(le) (rate(envoy_cluster_upstream_rq_time_bucket{kubernetes_namespace=~"$namespace", local_cluster=~"$service"}[5m])))
+```
+
+```promql
+histogram_quantile(0.999, sum by(le) (rate(envoy_cluster_upstream_rq_time_bucket{kubernetes_namespace=~"$namespace", local_cluster=~"$service"}[5m])))
+```
+
+### Total TCP inbound and outbound bytes
+
+**Description:** This time series shows the total number of inbound and outbound TCP bytes for services within the mesh. It provides visibility into the data transfer patterns and volume between services.
+
+```promql
+sum(rate(envoy_tcp_downstream_cx_rx_bytes_total{}[10m])) by (local_cluster)
+```
+
+### Total TCP inbound and outbound bytes buffered
+
+**Description:** This metric tracks the amount of TCP traffic buffered during inbound and outbound communications. It helps in identifying whether there is any potential latency caused by buffer overflow or congestion.
```

Comment:
```suggestion
**Description:** This metric tracks the amount of TCP traffic buffered during inbound and outbound communications. It helps in identifying whether there is any potential latency caused by packet buffering or congestion.
```
"Buffer overflow" has a specific meaning that I don't think is relevant here. I suggested an alternate wording.